### ISSUE TITLE
F filtering with __any__ option in the variable

### ISSUE THREAD

**Issue by Trasmontinho** [nornir-automation/nornir#607]
Hello, 

I'm facing a problem with the F filtering function. 

I've defined in the hosts data parameter a site. For personal purpose I'm filtering the sites with a list because sometimes I need to apply similar stuff in multiple sites.

In those site I've defined names quiet similar like 
r1:
    hostname: r1
    groups: 
        - blabla
    platform: cisco_nxos
    data:
        site: site1_client1
r2:
    hostname: r2
    groups: 
        - blabla
    platform: cisco_nxos
    data:
        site: site1

when i'm trying the command :
inventory_task = nr.filter(F(site__any=["site1"]))

I've got in inventory_task the two sites r1 & r2. I was expecting to match the exact value of each element of the list which is not the case.

Is it the expected behavior ? if it's not is there a way to fix it please ?

Thanks for your quick answer.

Regards, 

Sam

**Comment by dbarrosop** [comment]
Please, format the issue properly so the example can be followed. Also, make sure the example is complete and can be used to reproduce the issue you are describing. Thanks!

**Comment by Trasmontinho** [comment]
Hello @dbarrosop ,

Sorry I've been a bit quick I'll detail it .

Sam

**Comment by Trasmontinho** [comment]
Ok I'v attached a simple example that show my issue.
[nornir_issue.zip](https://github.com/nornir-automation/nornir/files/5502748/nornir_issue.zip)
 
So basically, I've got a dynamic list of devices list and sites issued of a magic done before based on requirements of the enduser of the script (for the purpose of the example I've just fixed the list of devices and site statically). 

In order to identify on a site the specific client where he wants to act I've put in the name of the site the name of the service to be quicker (I know that's ugly but for my need it fills the requirement :-)).

So basically in the example  you'll see what I've tested.

Filtering at the first the inventory with the list of the hosts and at the second stage a filtering based on a list of sites. But  as you can see in the example , 

the inventory hosts is not filtering as I was expecting : 
{'r1': Host: r1}
but gives me the output below : 
{'r1': Host: r1, 'r2': Host: r2, 'r3': Host: r3}

I don't know if it is "work as designed" or it's an issue (that's why I'm opening the issue). :-)

Thanks  @dbarrosop for your time on it.

Regards, 
Sam

**Comment by dbarrosop** [comment]
Sorry, I tend to not open zip files from unknown sources. What is the content of `groups.yaml`?

P.S: Please, format your messages with markdown

**Comment by Trasmontinho** [comment]
Hello @dbarrosop,

Ok sure. 
let me put below as plain text each file :
#config.yaml file
## Files are located in an inventory directory
inventory:
    plugin: SimpleInventory
    options:
        host_file: "inventory/hosts.yaml"
        group_file: "inventory/groups.yaml"
        defaults_file: "inventory/defaults.yaml"
runner:
    plugin: threaded
    options:
        num_workers: 10


# defaults.yaml file
username: admin
password: admin123

#groups.yaml file
## groups generated for further uses
datacenter: {}
wan: {}
campus: {}

#hosts.yaml file

r1:
    hostname: r1
    groups: 
        - datacenter
    platform: cisco_ios
    data:
        site: site1
r2:
    hostname: r2-public
    platform: cisco_nxos
    groups: 
        - wan
    data:
        site: site1_client_a
r3:
    hostname: r3
    groups: 
        - campus
    platform: cisco_ios
    data:
        site: site1_client_b

#test.py file
from nornir import InitNornir
from nornir.core.filter import F

nr = InitNornir("./inventory/config.yaml")
# Filter on all the hosts the one matching the list of sites defined by the enduser
inventory_task = nr.filter(F(hostname__any=["r1","r2","r3"])).filter(F(site__any=["site1"]))
print(inventory_task.inventory.hosts)
# expected result : {'r1': Host: r1}
#result obtained : {'r1': Host: r1, 'r2': Host: r2, 'r3': Host: r3}
inventory_task = nr.filter(F(hostname__any=["r1","r2","r3"])).filter(F(site__any=["site1","site1_client_a"]))
print(inventory_task.inventory.hosts)
# expected result : {'r1': Host: r1,  'r2': Host: r2,}
# obtained result : {'r1': Host: r1, 'r2': Host: r2, 'r3': Host: r3}

Regards, 

Sam

**Comment by Trasmontinho** [comment]
Sorry for the multiple updates maybe received I'm a beginner in github uses :-D
Regards, 

Sam

**Comment by dbarrosop** [comment]
So that wasn't supported as a matter of fact, it was semi working because unfortunately python treats strings as if they were lists. Anyway, I opened #608 to fix it. Feel free to test.

**Comment by Trasmontinho** [comment]
Thanks I test it and give you the feedback.

Regards, 
Sam

**Comment by Trasmontinho** [comment]
It works perfectly ! 👍 

Thanks for the fix @dbarrosop 

Last word, Thanks to you for such a great tool :-)

Regards, 

Sam
